### PR TITLE
Device emulator with TLS support

### PR DESCRIPTION
**Pull request by hvjunior**
This pull request implements TLS support for Device Emulator.

### CODE REVIEW

File: app/publishers/mqtt.js [hunk 1]
```diff
@@ -1,28 +1,62 @@
...
+    if ((mqttConfig.secure != undefined) && (mqttConfig.secure == true)) {
+      // Read TLS configuration
+      options.key = fs.readFileSync(mqttConfig.tls.key, 'utf8');
+      options.cert = fs.readFileSync(mqttConfig.tls.cert, 'utf8');
+      options.ca = [];
+      for (var i = 0; i < mqttConfig.tls.ca.length; i++) {
+        options.ca.push(fs.readFileSync(mqttConfig.tls.ca[i].name, 'utf8'));
+      }
+      // This should be removed from here ASAP
+      options.passphrase = 'cpqdiot2017';
```
Comment (mmagr) [inline app/publishers/mqtt.js]: This sounds like it could come from the configuration file no?

File: certs/ca.crt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN CERTIFICATE-----
```
Comment (mmagr) [inline certs/ca.crt]: This shouldn't be here

File: certs/iotagent.crt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+-----BEGIN CERTIFICATE-----
```
Comment (mmagr) [inline certs/iotagent.crt]: This shouldn't be here

File: certs/iotagent.key [hunk 4]
```diff
@@ -0,0 +1,15 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (mmagr) [inline certs/iotagent.key]: This most definitely shouldn't be here.

**Comment by giovannicuriel** [comment]
It's fine for me.

**Pull request merged**